src/transformers/models/kosmos2/processing_kosmos2.py

```diff
@@ -211,7 +211,9 @@ def __call__(
                 image_embeds_position_mask.append(mask)
 
             if isinstance(text, list):
-                sorted_length = sorted([(idx, len(x)) for idx, x in enumerate(text_encoding.input_ids)])
+                sorted_length = sorted(
+                    [(idx, len(x)) for idx, x in enumerate(text_encoding.input_ids)], key=lambda x: x[-1]
```

Comment:
Here my intention was to sort with the length, not the index. However, I forgot to specify the `key` so it did not do what I expected.

Reply:
Note that we do have tests regarding the batch mode - but the examples I used have the longest sequence as the last example in a batch, so it didn't detect this issue 😭

Reply:
Can we update the test so that it would have been caught?

Reply:
OK!